File: src/main/webapp/homeapp/src/app/rpi-backlight/rpi-backlight.service.ts
import {Injectable} from "@angular/core";
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs/Observable";
import {RpiBackLight} from "./rpi-backlight.model";

@Injectable()
export class RpiBackLightService {
  url = '/api/rpi-backlight';

  constructor(private httpClient: HttpClient) {

  }

  getBackLightIntensity(): Observable<number> {
    return this.httpClient.get(this.url);
  }

  setBackLightIntensity(rpiBackLight:RpiBackLight): Observable<RpiBackLight> {
    return this.httpClient.post(this.url, rpiBackLight);
  }
}

File: src/main/java/com/jorgma/homeapp/rpi/service/RpiBackLightService.java
package com.jorgma.homeapp.rpi.service;

import com.jorgma.homeapp.rpi.dao.RpiBackLight;

import java.io.IOException;

/**
 * Created by jorgma on 2018-03-24.
 */
public interface RpiBackLightService {
    int getBackLightIntensity();

    void setBackLightIntensity(RpiBackLight rpiBackLight) throws IOException;
}
